source/fundamentals/crud/read-operations/retrieve.txt

```diff
@@ -0,0 +1,242 @@
+.. _charp-retrieve:
+
+=============
+Retrieve Data
+=============
+
+.. default-domain:: mongodb
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+Overview
+--------
+
+In this guide, you can learn how to retrieve data from your MongoDB
+collections using the ``Find()`` method.
+
+Sample Data
+~~~~~~~~~~~
+
+The examples in this guide use the ``sample_restaurants.restaurants`` collection
+from the :atlas:`sample datasets </sample-data>` provided by Atlas. See the 
+:ref:`<csharp-quickstart>` to learn how to create a free MongoDB cluster and 
+load this sample data.
+
+.. literalinclude:: ../../../includes/code-examples/Restaurant.cs
+   :language: csharp
+   :copyable:
+   :dedent:
+
+.. _csharp-retrieve-find:
+
+Find Operation
+--------------
+
+You can use the ``Find()`` method to retrieve documents from a collection. 
+The ``Find()`` method takes a **query filter**, that specifies your
+search criteria, and returns all matching documents as an ``IFindFluent`` interface.
+
+Find One Document
+~~~~~~~~~~~~~~~~~
+
+To find a single document in a collection, pass a query filter that specifies the
+criteria of the document you want to find, then append the ``FirstOrDefault`` or 
+``FirstOrDefaultAsync`` method. If more than one document match the query
+filter these methods return the *first* matching document.
```

Comment:
Question/Suggestion: What is the first matching document? What order is used if no sort criteria is specified? Could a link be provided to learn more about this at the end of the section?

Reply:
I changed the wording of this to hopefully make it a little more clear. I'm hesitant to go into too much more detail on the specifics of the FirstOrDefault functionality (especially talking about sorting and things of that nature) on this page since I think there are a lot of details to it that could muddy the focus of this page.
I think the answer to "What is the first matching document" is largely depending on the query as well, which makes it hard to detail here.

Reply:
I like that, I think it's a good middle ground of explanation